src/service/requests/useUpdateUserRequest/index.ts

```diff
@@ -0,0 +1,46 @@
+import { AxiosError } from 'axios';
+import { useState } from 'react';
+import api from '../../api';
+import { TUpdateUserErrorResponse, TUpdateUserRequestBody } from './types';
+
+const useUpdateUserRequest = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string>();
+  const [isSuccess, setIsSuccess] = useState(false);
+
+  const updateUser = async (user: TUpdateUserRequestBody) => {
+    setIsLoading(true);
+    setIsSuccess(false);
+    setError(undefined);
+
+    try {
+      await api.patch(`/user`, { user });
```

Comment:
Aqui tem que passar os campos diretamente e não um campo `user` com os campos:

```suggestion
      await api.patch(`/user`, user);
```